### PR TITLE
Add Upload file

### PR DESCRIPTION
**Pull request by StKostyantin**
This PR is implemented upload module

### CODE REVIEW

File: modules/upload/client-react/containers/FileOperations.native.tsx [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+    return (
+      <ActionSheetProvider>
+        <Component {...props} />
+      </ActionSheetProvider>
+    );
+  };
+  return ActionSheet;
+};
+
+class FileOperations extends React.Component<FileOperationsProps, FileNativeOperationsState> {
```
Comment (evgenijolshanskij) [inline modules/upload/client-react/containers/FileOperations.native.tsx]: We should stop using `class`-based components in favor of functional ones.

File: modules/upload/server-ts/FileSystemStorage.ts [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+          reject(err);
+        }
+
+        resolve(stats);
+      });
+    });
+  }
+}
+
+export default new FileSystemStorage();
```
Comment (evgenijolshanskij) [inline modules/upload/server-ts/FileSystemStorage.ts]: This line creates a new instance of the `FileSystemStorage` every time it is being called.

File: modules/upload/client-react/index.tsx [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+import { Route, NavLink } from 'react-router-dom';
+import { translate, TranslateFunction } from '@restapp/i18n-client-react';
+import { MenuItem } from '@restapp/look-client-react';
+import ClientModule from '@restapp/module-client-react';
+
+import Upload from './containers/Upload';
+import resources from './locales';
+import uploadReducer from './reducers';
+
+const NavLinkWithI18n = translate('upload')(({ t }: { t: TranslateFunction }) => (
```
Comment (evgenijolshanskij) [inline modules/upload/client-react/index.tsx]: The `index` file should be clean and doesn't contain any components.

File: modules/upload/client-react/actions/uploadFiles.ts [hunk 4]
```diff
@@ -0,0 +1,23 @@
+import axios from 'axios';
+import { ActionType } from '../reducers';
+import { ActionFunction } from '.';
+
+const UPLOAD_FILES: ActionFunction = (files: any) => ({
```
Comment (evgenijolshanskij) [inline modules/upload/client-react/actions/uploadFiles.ts]: I have never seen functions to be called like this. Could you please give me a link to this convention.

File: modules/upload/server-ts/FileSystemStorage.ts (hunk 2 above)
Comment (StKostyantin) [reply modules/upload/server-ts/FileSystemStorage.ts]: Fixed

File: modules/upload/client-react/index.tsx (hunk 3 above)
Comment (StKostyantin) [reply modules/upload/client-react/index.tsx]: Fixed

File: modules/upload/client-react/actions/uploadFiles.ts (hunk 4 above)
Comment (StKostyantin) [reply modules/upload/client-react/actions/uploadFiles.ts]: Fixed

**Pull request closed without merging**